wgpu-core/src/device/life.rs

```diff
@@ -128,35 +128,65 @@ impl<A: hal::Api> NonReferencedResources<A> {
     }
 
     unsafe fn clean(&mut self, device: &A::Device) {
-        for raw in self.buffers.drain(..) {
-            device.destroy_buffer(raw);
+        {
+            profiling::scope!("destroy_buffers");
```

Comment:
let's guard this with `if !self.buffers.is_empty()` and everything below the same way